index.js

```diff
@@ -315,6 +321,19 @@ function getAuth (node) {
   }
 }
 
+function isHttpConnection (node) {
+  if (Array.isArray(node)) {
+    for (const n of node) {
+      if (new URL(n).protocol === 'http:') {
+        return true
+      }
+    }
+    return false
```

Comment:
__optional nit:__ (a bit easier to read, but it's more of a personal preference)
```suggestion
    return node.some((n) => new URL(n).protocol === 'http:')
```